packages/nimble-components/src/table/template.ts

```diff
@@ -1,32 +1,53 @@
-import { html, repeat } from '@microsoft/fast-element';
+import {
+    ElementsFilter,
+    html,
+    repeat,
+    slotted,
+    when
+} from '@microsoft/fast-element';
 import { DesignSystem } from '@microsoft/fast-foundation';
 import type { Table } from '.';
 import type { TableRecord } from './types';
 import { TableHeader } from './components/header';
 import { TableRow } from './components/row';
+import { TableColumn } from '../table-column/base';
+
+const isTableColumn = (): ElementsFilter => {
+    const filter: ElementsFilter = (
+        value: Node,
+        _: number,
+        __: Node[]
+    ): boolean => {
+        return value instanceof TableColumn;
+    };
+    return filter;
+};
 
 // prettier-ignore
 export const template = html<Table>`
     <template role="table">
         <div class="table-container">
             <div role="rowgroup" class="header-container">
                 <div class="header-row" role="row">
-                    ${repeat(x => x.columnHeaders, html<string>`
+                    ${repeat(x => x.columns, html<TableColumn>`
                         <${DesignSystem.tagFor(TableHeader)} class="header">
-                            ${x => x}
+                            ${x => x.textContent}
                         </${DesignSystem.tagFor(TableHeader)}>
                     `)}
                 </div>
             </div>
             <div class="table-viewport" role="rowgroup">
-                ${repeat(x => x.data, html<TableRecord>`
-                    <${DesignSystem.tagFor(TableRow)}
-                        :data="${x => x}"
-                        :columns="${(_, c) => (c.parent as Table).columns}"
-                    >
-                    </${DesignSystem.tagFor(TableRow)}>
+               ${when(x => x.columns.length > 0, html<Table>`
+                    ${repeat(x => x.data, html<TableRecord>`
```

Comment:
same as above, can include the parent type in the template so that we can avoid the `as Table`

Reply:
Done.